tests/scalers/nsq/nsq_test.go

```diff
@@ -0,0 +1,214 @@
+	ScaledObjectName string
+	JobName          string
+	MinReplicas      int
+	MaxReplicas      int
+	TopicName        string
+	ChannelName      string
+	MessageCount     int
+}
+
+func TestNSQScaler(t *testing.T) {
+	kc := GetKubernetesClient(t)
+
+	t.Cleanup(func() {
+		data, templates := getTemplateData()
+		uninstallNSQ(t)
+		KubectlDeleteWithTemplate(t, data, "jobTemplate", jobTemplate)
+		DeleteKubernetesResources(t, testNamespace, data, templates)
+	})
+
+	installNSQ(t, kc)
+
+	data, templates := getTemplateData()
+	CreateKubernetesResources(t, kc, testNamespace, data, templates)
+
+	require.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 0, 60, 1),
+		"Replica count should start out as 0")
+
+	testActivation(t, kc, data)
+	testScaleOut(t, kc, data)
+	testScaleIn(t, kc)
+}
+
+func installNSQ(t *testing.T, kc *kubernetes.Clientset) {
+	t.Log("--- installing NSQ ---")
+	CreateNamespace(t, kc, nsqNamespace)
+
+	_, err := ExecuteCommand("which helm")
+	require.NoErrorf(t, err, "nsq test requires helm - %s", err)
+
+	_, err = ExecuteCommand(fmt.Sprintf("helm repo add nsqio %s", nsqHelmRepoURL))
+	require.NoErrorf(t, err, "error while adding nsqio helm repo - %s", err)
+
+	_, err = ExecuteCommand(fmt.Sprintf("helm install nsq nsqio/nsq --namespace %s --set nsqadmin.enabled=false --wait", nsqNamespace))
+	require.NoErrorf(t, err, "error while installing nsq - %s", err)
+}
+
+func uninstallNSQ(t *testing.T) {
+	t.Log("--- uninstalling NSQ ---")
+	_, err := ExecuteCommand(fmt.Sprintf("helm uninstall nsq --namespace %s", nsqNamespace))
+	require.NoErrorf(t, err, "error while uninstalling nsq - %s", err)
+	DeleteNamespace(t, nsqNamespace)
+}
+
+func getTemplateData() (templateData, []Template) {
+	return templateData{
+			TestNamespace:    testNamespace,
+			NSQNamespace:     nsqNamespace,
+			DeploymentName:   deploymentName,
+			JobName:          jobName,
+			ScaledObjectName: scaledObjectName,
+			MinReplicas:      minReplicas,
+			MaxReplicas:      maxReplicas,
+			TopicName:        topicName,
+			ChannelName:      channelName,
+		}, []Template{
+			{Name: "deploymentTemplate", Config: deploymentTemplate},
+			{Name: "scaledObjectTemplate", Config: scaledObjectTemplate},
+		}
+}
+
+func testActivation(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- testing activation ---")
+
+	data.MessageCount = 5
+	KubectlReplaceWithTemplate(t, data, "jobTemplate", jobTemplate)
+
+	AssertReplicaCountNotChangeDuringTimePeriod(t, kc, deploymentName, testNamespace, 0, 60)
+}
+
+func testScaleOut(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- testing scale out ---")
+
+	data.MessageCount = 1 // 5 already published + 1 > activationDepthThreshold
+	KubectlReplaceWithTemplate(t, data, "jobTemplate", jobTemplate)
+
+	require.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 1, 60, 1),
+		"replica count should be 1 after 1 minute")
+}
```

Comment:
We should increase the queue to scale from 1 to at least 2 to verify that the scaling phase also works. My suggestion at this point is setting `maxReplicaCount:2` and updating the consumer to introduce a sleep of 1 second or so after getting each message. This allows to enqueue 30-60 extra messages, ensuring that the workload scales to 2 instances

Reply:
I was able to modify the test image here [kedacore/test-tools/#184](https://github.com/kedacore/test-tools/pull/184). The e2e tests should be better now 👍